src/internal/packager/common.go

```diff
@@ -163,6 +169,32 @@ func HandleIfURL(packagePath string, shasum string, insecureDeploy bool) (string
 	return localPackagePath, tempPath.clean
 }
 
+func handleSgetPackage(sgetPackagePath string) (string, func()) {
+	sgetPackageName := strings.Split(sgetPackagePath, "sget://")[1]
+	// Write the package to a local file
+	tempPath := createPaths()
+
+	// Create the file
+	localPackagePath := tempPath.base + sgetPackageName
+	destinationFile, err := os.Create(localPackagePath)
+	if err != nil {
+		message.Fatal(err, "Unable to create the destination file")
+	}
+	defer destinationFile.Close()
+
+	// If this is a DefenseUnicorns package, use an internal sget public key
+	if strings.HasPrefix(sgetPackagePath, "sget://defenseunicorns") {
+		os.Setenv("DU_SGET_KEY", config.SGetPublicKey)
+		config.DeployOptions.SGetKeyPath = "env://DU_SGET_KEY"
```

Comment:
Cosign can load the public key as an environment variable now. This must be new since when we last looked at it!

https://github.com/sigstore/cosign/blob/29361993239764ae63c3017198cc80ff5816c08f/pkg/blob/load.go#L46